1_download_dialogs_data.py

```diff
@@ -1,300 +1,110 @@
-import os
 import argparse
-from typing import Dict
-
-import pandas as pd
-import logging
-import json
+from typing import Callable
 
 import telethon
 
-from utils.utils import init_config, read_dialogs
-
-REACTIONS_LIMIT_PER_MESSAGE = 100
+from telegram_data_downloader.dict_types.dialog import DialogMetadata, DialogType
+from telegram_data_downloader.factory import (
+    create_json_dialog_reader_writer,
+    create_message_downloader,
+    create_telegram_client,
+)
 
 
 def init_args():
-    """
-    Initialize arguments
-
-    :return: argparse.Namespace
-    """
-    parser = argparse.ArgumentParser(description="Step #2.Download dialogs")
+    parser = argparse.ArgumentParser(description="Step #2. Download dialogs")
 
     parser.add_argument(
-        "--dialogs_ids",
+        "--dialog-ids",
         nargs="+",
         type=str,
         help="id(s) of dialog(s) to download, -1 for all",
         required=True,
     )
     parser.add_argument(
-        "--dialog_msg_limit",
+        "--dialog-msg-limit",
         type=int,
         help="amount of messages to download from a dialog, -1 for all",
         default=10000,
     )
-    parser.add_argument(
-        "--config_path",
-        type=str,
-        help="path to config file",
-        default="config/config.json",
-    )
-    parser.add_argument("--debug_mode", type=int, help="Debug mode", default=0)
-    parser.add_argument("--session_name", type=str, help="session name", default="tmp")
-    parser.add_argument("--skip_private", action="store_true")
-    parser.add_argument("--skip_groups", action="store_true")
-    parser.add_argument("--skip_channels", action="store_true")
+    parser.add_argument("--session-name", type=str, help="session name", default="tmp")
+    parser.add_argument("--skip-private", action="store_true")
+    parser.add_argument("--skip-groups", action="store_true")
+    parser.add_argument("--skip-channels", action="store_true")
 
     return parser.parse_args()
 
 
-def dialogs_id_input_handler(
-    input_id_lst, is_dialog_type_accepted, dialog_list="data/dialogs"
-):
-    """
-    Functions handles input_id_lst depending on the input
-
-    :param input_id_lst: list of all dialogs from command prompt
-    :param is_dialog_type_accepted:
-    :param dialog_list: list of all dialogs in meta directory
-    :return:
-    """
+def filter_input_dialogs(
+    input_id_lst: list[str],
+    is_dialog_type_accepted: dict[DialogType, bool],
+    dialog_list: list[DialogMetadata],
+) -> list[DialogMetadata]:
+    condition: Callable[[DialogMetadata], bool] = lambda dialog: True  # noqa: E731
 
     if input_id_lst[0] == "-1":
-        return [
-            dialog["id"]
-            for dialog in dialog_list
-            if is_dialog_type_accepted[dialog["type"]]
-        ]
+        condition = condition  # don't change anything
```

Comment:
why do we need this?

Reply:
basically, for exiting early, but reassignment is indeed excessive